frb_codegen/src/config.rs

```diff
@@ -78,30 +79,109 @@ pub struct Opts {
     pub manifest_path: String,
     pub dart_root: Option<String>,
     pub build_runner: bool,
+
+    // due to conflict of dart api of multiple blocks, keep it temporarily
     pub exclude_sync_execution_mode_utility: bool,
 }
 
-pub fn parse(raw: RawOpts) -> Opts {
-    let rust_input_path = canon_path(&raw.rust_input);
-
-    let rust_crate_dir = canon_path(&raw.rust_crate_dir.unwrap_or_else(|| {
-        fallback_rust_crate_dir(&rust_input_path)
-            .unwrap_or_else(|_| panic!("{}", format_fail_to_guess_error("rust_crate_dir")))
-    }));
-    let manifest_path = {
-        let mut path = std::path::PathBuf::from_str(&rust_crate_dir).unwrap();
-        path.push("Cargo.toml");
-        path_to_string(path).unwrap()
-    };
-    let rust_output_path = canon_path(&raw.rust_output.unwrap_or_else(|| {
-        fallback_rust_output_path(&rust_input_path)
-            .unwrap_or_else(|_| panic!("{}", format_fail_to_guess_error("rust_output")))
-    }));
-    let class_name = raw.class_name.unwrap_or_else(|| {
-        fallback_class_name(&*rust_crate_dir)
-            .unwrap_or_else(|_| panic!("{}", format_fail_to_guess_error("class_name")))
+pub fn parse(raw: RawOpts) -> Vec<Opts> {
+    // rust input path(s)
+    let mut rust_input_paths = raw.rust_input.clone();
+    rust_input_paths.retain(|each_path| !each_path.trim().is_empty());
+    let rust_input_paths = rust_input_paths
+        .iter()
+        .map(|each_path| canon_path(each_path))
+        .collect::<Vec<_>>();
+    assert!(
+        rust_input_paths.len() >= 1,
+        "rust input(s) should have at least 1 path"
+    );
+
+    // dart output path(s)
+    let mut dart_output_paths = raw.dart_output.clone();
+    dart_output_paths.retain(|each_path| !each_path.trim().is_empty());
+    let dart_output_paths = dart_output_paths
+        .iter()
+        .map(|each_path| canon_path(each_path))
+        .collect::<Vec<_>>();
+    assert!(
```

Comment:
Moreover, you are duplicating code between rust_input_paths and dart_output_paths above. Can we extract into a new method?

Reply:
Fixed in 9e38c2995f796d3e5c35b6bbaa631d62645325dc, with new function named `get_valid_canon_paths`.